src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderActivitiesResourceTest.java

```diff
@@ -68,5 +69,76 @@ public void successfulRequest_page() {
             assertThat(jsonContent).extractingJsonPathNumberValue("$.totalPages").isEqualTo(3);
             assertThat(jsonContent).extractingJsonPathNumberValue("$.totalElements").isEqualTo(5);
         }
+
+        @Test
+        public void noRole() {
+            final var entity = createHttpEntity(validToken(List.of()), null);
+
+            final var response = testRestTemplate.exchange(
+                "/api/offender-activities/1234/activities-history?earliestEndDate=2021-01-01",
+                HttpMethod.GET,
+                entity,
+                String.class);
+
+            assertThatStatus(response, HttpStatus.FORBIDDEN.value());
+        }
+    }
+
+    @Nested
+    class GetHistoricalAttendances {
+        @Test
+        public void successfulRequest_returnsCorrectData() {
+            final var entity = createHttpEntity(validToken(List.of("ROLE_SYSTEM_USER")), null);
+            final var today = LocalDate.now().format(DateTimeFormatter.ISO_LOCAL_DATE);
+
+            final var response = testRestTemplate.exchange(
+                "/api/offender-activities/A1234AB/attendance-history?fromDate=2017-01-01&toDate=" + today + "&page=1&size=2&sort=eventId,desc",
+                HttpMethod.GET,
+                entity,
+                String.class);
+
+            assertThatJsonFileAndStatus(response, HttpStatus.OK.value(), "offender-attendence-history.json");
```

Comment:
I think these tests would read better if you insert the data in the test setup then use JSON asserts inline to assert on the data returned, else you have to scan the seed that and that file to understand what the API is doing

Reply:
Agree - the seed data has become very 'entangled' with multiple tests. Will address next time.